### PR TITLE
Multi statement, Attach, Detach, Use statements support

### PR DESCRIPTION
**Pull request by hmeriann**
* Added multi-statement generation to the fuzzyduck.cpp;
* StatementGenerator::RandomValue() and StatementGenerator::RandomPercentage() became public to be used in fuzzyduck.cpp

### CODE REVIEW

**Comment by Tmonster** [comment]
Edit: Nevermind, this is just for multi statements, and the PR to add attach & use will come later

File: src/statement_generator.cpp [hunk 1]
```diff
@@ -4,22 +4,30 @@
...
 #include "duckdb/parser/statement/delete_statement.hpp"
+#include "duckdb/parser/statement/detach_statement.hpp"
 #include "duckdb/parser/statement/insert_statement.hpp"
+#include "duckdb/parser/statement/multi_statement.hpp"
 #include "duckdb/parser/statement/select_statement.hpp"
+#include "duckdb/parser/statement/set_statement.hpp"
 #include "duckdb/parser/statement/update_statement.hpp"
 #include "duckdb/parser/tableref/list.hpp"
 
+#define TESTING_DIRECTORY_NAME "duckdb_unittest_tempdir"
```
Comment (Tmonster) [inline src/statement_generator.cpp:29]: Now that duckdb_sqlsmith is an extension and has duckdb as a submodule is it possible to use the `TESTING_DIRECTORY_NAME` declaration from `test/helpers/test_helpers.cpp`?

Comment (Tmonster) [reply src/statement_generator.cpp:29]: you might have to include `duckdb/test/include/test_helpers.hpp`

**Comment by hmeriann** [comment]
@Mytherin could you please have a look at this PR?

**Review by Tmonster — APPROVED** [review]
Just the one comment. Otherwise Looks good to me from here

File: src/statement_generator.cpp (hunk 1 above)
Comment (hmeriann) [reply src/statement_generator.cpp:29]: In GenerateAttachInfo() it checks if directory exists and creates it if it doesn't. Maybe better to use a different name for sqlsmiths testing directory? Something like sqlsmith_unittest_tempdir maybe?

Comment (hmeriann) [reply src/statement_generator.cpp]: Defined new TESTING_DIRECTORY_NAME, also put it in the .gitignore

File: src/statement_generator.cpp [hunk 2]
```diff
@@ -109,6 +145,76 @@ unique_ptr<CreateStatement> StatementGenerator::GenerateCreate() {
...
+
+//===--------------------------------------------------------------------===//
+// Generate Attach Info
+//===--------------------------------------------------------------------===//
+
+unique_ptr<AttachInfo> StatementGenerator::GenerateAttachInfo() {
+	auto info = make_uniq<AttachInfo>();
+	auto fs = FileSystem::CreateLocal();
+	// check if the directory exists
+	if (!fs->DirectoryExists(string("../") + TESTING_DIRECTORY_NAME)) {
```
Comment (Tmonster) [inline src/statement_generator.cpp:207]: remove the "../" these cause the testing directory to end up in a weird place

**Review by Tmonster — CHANGES_REQUESTED** [review]
So I think we should stick with "duckdb_unittest_tempdir". This directory is already in the `.gitignore`. Also, if someone builds sqlsmith in the duckdb directory or in the sqlsmith directory, the databases attached by the attach and use will automatically be in the `.gitignore`

File: src/statement_generator.cpp (hunk 2 above)
Comment (hmeriann) [reply src/statement_generator.cpp]: Removed - now it puts files generated during testing to the proper `duckdb_unittest_tempdir`

**Review by Tmonster — APPROVED** [review]
lgtm 👍 . Also agree with Mark that we can just directly refer to persistent databases in the future instead of using `USE`

**Comment by Tmonster** [comment]
Thanks!

**Pull request merged**